### DISCUSSION TITLE
Catching "unused primitive" warnings before running simulation

### DISCUSSION THREAD

**Discussion by 0xCoto** [thliebig/openEMS-Project#93] [Q&A Usage]
The "unused primitive" warning (which is usually (always?) caused by a coarse mesh) is one of the most common warnings in openEMS, and can often lead to inaccurate results if the mesh is not revised. Is it somehow possible to quickly catch these warnings in advance without having to run the simulation?

**Comment by thliebig** [comment]
No unfortunately not. The discretization process is very complex... The only thing you can do is tell openEMS to not start the simulation and only do the pre-processing, but that does not give any gain if everything checks out and you have to do it all again...

**Reply by thliebig** [reply]
Well of course some trivial checks would be possible, like is an object completely outside the mesh domain. 
But I guess that is not what you are getting at...

**Reply by 0xCoto** [reply]
Yeah, I'm basically trying to have a separate quick mesh validator before running the simulation and catch non-obvious unused primitive warnings in advance (which should often be interpreted as errors instead of being ignored and proceeding to simulate anyway). I guess `--no-simulation` isn't currently exposed in the [Python interface](https://github.com/thliebig/openEMS/blob/638d87590631dc0a1a08cabd1c741a4498a81851/python/openEMS/openEMS.pyx#L450-L451), so I might make a PR (and also include `--engine` while I'm at it).
